packages/backend-storage/src/construct.ts

```diff
@@ -83,6 +83,8 @@ export class AmplifyStorage
           ],
         },
       ],
+      autoDeleteObjects: true,
+      removalPolicy: RemovalPolicy.DESTROY,
```

Comment:
Would the knob be `retainOnDelete?: boolean` with a default of false?

Reply:
something like that or `dataRetention: DataRetention.RETAIN/DELETE` with default `DELETE`. 
i.e. enum instead of boolean. in case we have DataRetention.REMOVE_EMPTY_BUCKETS_ONLY some day.

Reply:
my last comment made me thing that maybe we should apply YAGNI and see what asks are ;-)